src/Patches.cs

```diff
@@ -169,51 +196,78 @@ private static void RecordData(string line, string informationType, bool append)
 
-			if (container.m_LootTablePrefab != null) return container.m_LootTablePrefab.name;
+			if (container.m_LootTableData != null) return container.m_LootTableData.name;
 
 			return null;
 		}
 	}
 
-	[HarmonyPatch(typeof(PlayerManager), "GetInteractiveObjectDisplayText")]
-	internal class PlayerManager_GetInteractiveObjectDisplayText
+	[HarmonyPatch(typeof(PlayerManager), nameof(PlayerManager.GetInteractiveObjectUnderCrosshairs), new Type[] { typeof(float) })]
+	internal class GetInteractiveObjectUnderCrosshairs
 	{
-		public static void Postfix(PlayerManager __instance, ref string __result)
+		public static void Postfix(ref GameObject? __result)
 		{
-			if (__instance?.m_InteractiveObjectUnderCrosshair?.name is null) return;
-			if (__result is null || string.IsNullOrWhiteSpace(GameManager.m_ActiveScene)) return;
+			if (__result == null || string.IsNullOrWhiteSpace(GameManager.m_ActiveScene))
+			{
+				KeyTracker.ResetLooatingAt();
+				return;
+			}
+
+			if (Settings.options.useDeleteFunction && CustomInput.GetKeyDown(Settings.options.deleteKey))
+			{
+				UnityEngine.Object.Destroy(__result);
+				KeyTracker.ResetLooatingAt();
+				__result = null;
+				return;
+			}
+
+			if (KeyTracker.lookingAt == __result || KeyTracker.currentMode == GrabberMode.None)
+			{
+				return;
+			}
+
+			string outputMsg = string.Empty;
+
 			switch (KeyTracker.currentMode)
 			{
 				case GrabberMode.Name:
-					__result += "\nname = " + __instance.m_InteractiveObjectUnderCrosshair.name;
+					outputMsg = "\nname = " + __result.name;
 					break;
 				case GrabberMode.Position:
-					__result += "\nposition = " + FormatHelper.FormatVector(__instance.m_InteractiveObjectUnderCrosshair.transform.position);
+					outputMsg = "\nposition = " + FormatHelper.FormatVector(__result.transform.position);
 					break;
 				case GrabberMode.Rotation:
-					__result += "\nrotation = " + FormatHelper.FormatVector(__instance.m_InteractiveObjectUnderCrosshair.transform.rotation.eulerAngles);
+					outputMsg = "\nrotation = " + FormatHelper.FormatVector(__result.transform.rotation.eulerAngles);
 					break;
 				case GrabberMode.Scene:
-					__result += "\nscene = " + GameManager.m_ActiveScene;
+					outputMsg = "\nscene = " + GameManager.m_ActiveScene;
 					break;
 				case GrabberMode.LootTable:
-					Container container = __instance.m_InteractiveObjectUnderCrosshair.GetComponentInChildren<Container>();
+					Container container = __result.GetComponentInChildren<Container>();
 					if (container != null)
 					{
-						__result += "\nloottable = " + LootTableHelper.GetLootTableName(container);
+						outputMsg = "\nloottable = " + LootTableHelper.GetLootTableName(container);
+					}
+					else
+					{
+						return;
 					}
 					break;
 			}
-			if (Settings.options.useDeleteFunction && CustomInput.GetKeyDown(Settings.options.deleteKey))
+
+			if (!string.IsNullOrWhiteSpace(outputMsg))
```

Comment:
If the above change about the default switch case is implemented, this if check will always be true.